src/components/Stage/index.tsx

```diff
@@ -1,32 +1,54 @@
-import { useState } from 'react';
+import { useState, FC } from 'react';
 import Wordbox from '../Wordbox';
 import wordList from '../../word-list';
 import './style.css';
 
 // TODO: temporary disable function - remove next line when you start using it
 /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
 const generateWord = (size: number) => {
-  const sizeIndex = size === undefined
-    ? Math.floor(Math.random() * wordList.length)
-    : size - 3;
-  
+  const sizeIndex =
+    size === undefined ? Math.floor(Math.random() * wordList.length) : size - 3;
+
   if (sizeIndex < 0 || sizeIndex >= wordList.length) {
     return null;
   }
-  
+
   const words = wordList[sizeIndex];
   const wordIndex = Math.floor(Math.random() * words.length);
   return words[wordIndex];
 };
 
-const Stage = () => {
-  const [words] = useState<string[]>(['jahoda']);
+const Stage: FC = () => {
+  const [words, setWords] = useState<string[]>(['jahoda', 'atrapa', 'brynza']);
+  const [numOfErrors, setNumOfErrors] = useState<number>(0);
+
+  const handleFinish = () => {
+    const newWord = generateWord(6);
+    if (newWord !== null) {
+      const newArray = [...words];
+      newArray.shift();
+      newArray.push(newWord);
```

Comment:
Tohle funguje dobře, ale když bys chtěla, můžeš si to zkusit přepsat na immutable řešení (jak jste se učily [tady](https://kodim.cz/czechitas/react-2/lekce/hodnoty-stav/imutabilita)).